docs/adr/0004-framework.md

```diff
@@ -0,0 +1,68 @@
+# Use React Framework
+
+* Status: proposed
+* Deciders: Monitor team
+* Date: 2022-05-03
+* Technical Story: [MNTOR-1588](https://mozilla-hub.atlassian.net/browse/MNTOR-1588)
+
+## Context and Problem Statement
+
+The Monitor team is tasked with launching a premium product offering as soon as possible. The current architecture of Monitor is not suited for synchronous feature development and lacks the complex client-side state management premium will require. 
+
+
+## Decision Drivers <!-- optional -->
+
+* The expansion in Monitor premium requires a framework to manage increased complexity and client-side state management
+* Promote component sharing between S&P web-based products, Relay & Monitor
+* Allow developers to work on specific components and features without affecting the rest of the codebase (fewer merge conflicts, makes it easier break down tickets)
+* Unidirectional (one-way) data flow makes it easier to understand the flow of data in the application, which may be necessary when handling multiple states (especially for the data removal sequence)
+* Monitor Front-end engineers have extensive React experience
+
+## Considered Options
+
+1. Migrate framework to React and Next.js
+1. Leave as-is 
+1. Use a different framework
+
+## Decision Outcome
+
+TBD
+
+## Pros and Cons of the Options <!-- optional -->
+
+### Migrate framework to React and Next.js
+
+* Good, Component-based Architecture: This simplifies code maintenance by breaking complex UIs into reusable components.
+* Good, Virtual DOM: React's virtual DOM allows for efficient updates and rendering of the UI, which can improve performance and speed up development.
+* Good, Testing: Test components in isolation because of its modularity
+* Good, Accessibility: React provides ARIA support, focus management, and components are designed with accessibility best practices in mind.
+* Good, Shared knowledge: This approach shares the same structure as Firefox Relay. This would immediately allow the team to reuse and reference.
+	* Note that FxA came to a similiar decision: [FxA ADR: Refactor Subscription Platform frontend - with Next.js](https://github.com/mozilla/fxa/blob/main/docs/adr/0035-refactor-payments-frontend-with-nextjs.md) 
+* Good, Typescript: TypeScript's static typing enhances code robustness and readability by catching errors at compile-time and making code more self-explanatory.
+* Good, Industry Usage/Standard: React has a large and active community of developers, with many resources available, including libraries, tutorials, and support.
+* Good, Simpler Code: Hooks in React utilizes simpler code that implements similar functionalities faster and more effectively
+* Good, [fluent-react](https://github.com/projectfluent/fluent.js/tree/main/fluent-react) enables more dynamic localized texts and messages
+* Bad, Tooling Overhead: React's tooling and ecosystem can be complex, which can be a disadvantage for developers who prefer a more minimalistic approach.
```

Comment:
More pros:

- Avoid snowflake configs: we won't have to manually set up tooling and make sure they work well together, like https://github.com/mozilla/blurts-server/pull/3003 and https://github.com/mozilla/blurts-server/pull/2987.
- Automatic code splitting, avoiding e.g. cumulative layout shift, and decreating bundle sizes.

More cons:
- We give away some autonomy w.r.t. what tooling we can use and when we can use the latest versions.
- The client-side execution time of React is additional overhead.

Reply:
Added in 1714708a46edf08c1eea91fa751ddd86dbbe5cac